Assets/MRTK/Providers/OpenXR/Scripts/MicrosoftArticulatedHand.cs

```diff
@@ -76,20 +86,80 @@ public override void UpdateController(InputDevice inputDevice)
                 if (inputDevice.TryGetFeatureValue(CommonUsages.devicePosition, out Vector3 _))
                 {
                     base.UpdateController(inputDevice);
+
+                    // We've gotten device data from the platform, don't attempt to infer other input actions
+                    // from the hand joint data
+                    receivingDeviceInputs = true;
                 }
                 else
                 {
                     UpdateHandData(inputDevice);
 
+                    // Updating the Index finger pose right after getting the hand data
+                    // regardless of whether device data is present
                     for (int i = 0; i < Interactions?.Length; i++)
                     {
-                        switch (Interactions[i].AxisType)
+                        var interactionMapping = Interactions[i];
+                        switch (interactionMapping.InputType)
                         {
-                            case AxisType.SixDof:
-                                UpdatePoseData(Interactions[i], inputDevice);
+                            case DeviceInputType.IndexFinger:
+                                handDefinition?.UpdateCurrentIndexPose(Interactions[i]);
                                 break;
                         }
                     }
+
+                    // If we aren't getting device data, infer input actions, velocity, etc from hand joint data
+                    if (!receivingDeviceInputs)
+                    {
+                        for (int i = 0; i < Interactions?.Length; i++)
+                        {
+                            switch (Interactions[i].InputType)
+                            {
+                                case DeviceInputType.SpatialGrip:
+                                    MixedRealityPose currentGripPose = unityJointPoses[TrackedHandJoint.Palm];
```

Comment:
I think we may want to transform this slightly to match platform conventions. There's a slight rotation, which would basically be the inverse of https://github.com/microsoft/MixedRealityToolkit-Unity/blob/5d07c72ec0d62669bd5cdb7ab8b464831d46019a/Assets/MRTK/Providers/WindowsMixedReality/XRSDK/Controllers/WindowsMixedRealityXRSDKArticulatedHand.cs#L57-L58

which is used to get from grip pose to palm pose at 
https://github.com/microsoft/MixedRealityToolkit-Unity/blob/5d07c72ec0d62669bd5cdb7ab8b464831d46019a/Assets/MRTK/Providers/WindowsMixedReality/XRSDK/Controllers/WindowsMixedRealityXRSDKArticulatedHand.cs#L155-L158

Reply:
Done!